pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -180,60 +211,62 @@ func (r *Reconciler) podSpecChanged(oldPodSpec corev1.PodSpec, newPodSpec corev1
 	return false
 }
 
-func (r *Reconciler) runAccessCheck(ctx context.Context, src *v1.ApiServerSource) error {
+func (r *Reconciler) runAccessCheck(ctx context.Context, src *v1.ApiServerSource, namespaces []string) error {
 	if src.Spec.Resources == nil || len(src.Spec.Resources) == 0 {
 		src.Status.MarkSufficientPermissions()
 		return nil
 	}
 
-	user := "system:serviceaccount:" + src.Namespace + ":"
-	if src.Spec.ServiceAccountName == "" {
-		user += "default"
-	} else {
-		user += src.Spec.ServiceAccountName
-	}
-
 	verbs := []string{"get", "list", "watch"}
 	lastReason := ""
 
 	// Collect all missing permissions.
 	missing := ""
 	sep := ""
 
-	for _, res := range src.Spec.Resources {
-		gv, err := schema.ParseGroupVersion(res.APIVersion)
-		if err != nil {
-			return err
-		}
-		gvr, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Kind: res.Kind, Group: gv.Group, Version: gv.Version}) // TODO: Test for nil Kind.
-		missingVerbs := ""
-		sep1 := ""
-		for _, verb := range verbs {
-			sar := &authorizationv1.SubjectAccessReview{
-				Spec: authorizationv1.SubjectAccessReviewSpec{
-					ResourceAttributes: &authorizationv1.ResourceAttributes{
-						Namespace: src.Namespace,
-						Verb:      verb,
-						Group:     gv.Group,
-						Resource:  gvr.Resource,
-					},
-					User: user,
-				},
-			}
+	user := "system:serviceaccount:" + src.Namespace + ":"
+	if src.Spec.ServiceAccountName == "" {
+		user += "default"
+	} else {
+		user += src.Spec.ServiceAccountName
+	}
 
-			response, err := r.kubeClientSet.AuthorizationV1().SubjectAccessReviews().Create(ctx, sar, metav1.CreateOptions{})
+	for _, ns := range namespaces {
+		for _, res := range src.Spec.Resources {
+			gv, err := schema.ParseGroupVersion(res.APIVersion)
 			if err != nil {
 				return err
 			}
+			gvr, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Kind: res.Kind, Group: gv.Group, Version: gv.Version}) // TODO: Test for nil Kind.
+			missingVerbs := ""
+			sep1 := ""
```

Comment:
We can avoid doing the parsing + unsafe guess multiple times since it doesn't change per namespace